### PR TITLE
Add end_strings to SamplingParams

### PR DESCRIPTION
**Pull request by markelsanz14**
# What does this PR do ?
Makes end_strings part of SamplingParams and makes it configurable.

**Collection**: Text Generation Server

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/nlp/language_modeling/conf/megatron_gpt_inference.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ inference:
   repetition_penalty: 1.2  # The parameter for repetition penalty. 1.0 means no penalty.
   min_tokens_to_generate: 0  # The minimum length of the sequence to be generated.
   compute_logprob: False  # a flag used to compute logprob of all the input text, a very special case of running inference, default False
-
+  end_strings: ["<|endoftext|>", "<extra_id_1>"]  # generation will stop when one of these tokens is generated
```
Comment (MaximumEntropy) [inline examples/nlp/language_modeling/conf/megatron_gpt_inference.yaml:12]: I don't think we should add `<extra_id_1>` as part of the defaults here.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py [hunk 2]
```diff
@@ -375,6 +375,7 @@ def inference_step(self, dataloader_iter, batch_idx, mode, dataloader_idx=0):
             "add_BOS": False,
             "all_probs": False,
             "compute_logprob": False,
+            "end_strings": ["<|endoftext|>", "<extra_id_1>"],
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py:378]: There is some inconsistency between here and above about "default" end strings. Let us use only <|endoftext|> everywhere.

File: tests/collections/nlp/test_gpt_eval.py [hunk 3]
```diff
@@ -78,6 +78,7 @@ def test_gpt_eval(self):
             "add_BOS": True,
             "all_probs": False,
             "compute_logprob": False,
+            "end_strings": ["<|endoftext|>", "<extra_id_1>"],
```
Comment (MaximumEntropy) [inline tests/collections/nlp/test_gpt_eval.py:81]: Same here about defaults.

**Review by MaximumEntropy — CHANGES_REQUESTED** [review]
Minor comment about including `<extra_id_1>` as a default end string everywhere.

File: examples/nlp/language_modeling/conf/megatron_gpt_inference.yaml (hunk 1 above)
Comment (markelsanz14) [reply examples/nlp/language_modeling/conf/megatron_gpt_inference.yaml]: Removed.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py (hunk 2 above)
Comment (markelsanz14) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py]: Removed.

File: tests/collections/nlp/test_gpt_eval.py (hunk 3 above)
Comment (markelsanz14) [reply tests/collections/nlp/test_gpt_eval.py]: Removed.

**Review by MaximumEntropy — DISMISSED** [review]

**Review by yidong72 — APPROVED** [review]

**Review by ericharper — APPROVED** [review]
Re-approving.

**Pull request merged**